src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java

```diff
@@ -16,9 +22,23 @@
     private static final Logger LOGGER = LoggerFactory
         .getLogger(CompanyAccountsApplication.APPLICATION_NAME_SPACE);
 
+    @Autowired
+    private FilingService filingService;
+
     @GetMapping("/private/transactions/{transactionId}/company-accounts/{companyAccountId}/filings")
-    public ResponseEntity generateFiling() {
+    public ResponseEntity generateFiling(@PathParam("transactionId") String transactionId,
+        @PathParam("companyAccountId") String accountId, HttpServletRequest request) {
+
+        Filing filing = filingService.generateAccountFiling();
+        if (filing != null) {
+            return new ResponseEntity<>(Arrays.asList(filing), HttpStatus.OK);
+        }
+
+        final Map<String, Object> debugMap = new HashMap<>();
```

Comment:
We could put this into its own method so other methods could use it as it may get duplicated in the long run? However, I'm not fussy on this, more of a observation for future reference.

Reply:
it was in a method before in the previous pr. I will move it to separate method